src/components/settings/NumberSetting.tsx

```diff
@@ -62,45 +63,50 @@ export const NumberSetting = ({
     handleUpdate,
     showSlider,
     disabled = false,
+    handleLiveUpdate,
     listItemTextSx: sx,
 }: Props) => {
     const { t } = useTranslation();
 
     const [isDialogOpen, setIsDialogOpen] = useState(false);
     const [dialogValue, setDialogValue] = useState(value);
+    const [originalValue, setOriginalValue] = useState(value);
 
-    const closeDialog = useCallback(
-        (resetValue: boolean) => {
-            setIsDialogOpen(false);
-
-            if (resetValue) {
-                setDialogValue(value);
+    const updateValue = useCallback(
+        (newValue: number) => {
+            setDialogValue(newValue);
+            if (handleLiveUpdate) {
+                handleLiveUpdate(newValue);
             }
         },
-        [value],
+        [handleLiveUpdate],
     );
 
-    const closeDialogWithReset = useCallback(() => closeDialog(true), [closeDialog]);
-
-    const updateSetting = useCallback(
-        (newValue: number, shouldCloseDialog: boolean = true) => {
-            if (shouldCloseDialog) {
-                closeDialog(false);
-            }
+    const cancel = useCallback(() => {
+        setDialogValue(originalValue);
+        setOriginalValue(originalValue);
+        handleUpdate(originalValue);
+        setIsDialogOpen(false);
+    }, [originalValue, handleUpdate]);
 
-            const didValueChange = value !== newValue;
-            if (!didValueChange) {
-                return;
-            }
-
-            handleUpdate(newValue);
-        },
-        [value, handleUpdate, closeDialog],
-    );
+    const resetToDefault = useCallback(() => {
+        if (defaultValue !== undefined) {
+            setDialogValue(defaultValue);
+            setOriginalValue(defaultValue);
+            handleUpdate(defaultValue);
+            setIsDialogOpen(false);
+        }
+    }, [defaultValue, handleUpdate]);
 
-    useEffect(() => {
-        setDialogValue(value);
-    }, [value]);
+    const submit = () => {
+        const didValueChange = dialogValue !== originalValue;
```

Comment:
this should be moved to `handleUpdate` since `cancel` and `resetToDefault` value currently send a request for the same value

Reply:
Hmm, previously this was in the `updateSetting` function and moving this into the `handleUpdate` prop means that all existing uses of `NumberSetting` will need to be updated, is that what's wanted?

Reply:
you're right, it should be handled in this component and not the prop, I mistook this for `updateValue`

I think it would be best to e.g. add a `persist` parameter to `updateValue` and then call `handleUpdate` instead of `handleLiveUpdate` in case `true` gets passed
with this everything else has to call `updateValue` to be able to change it

then the `didValueChange` check could be move there to prevent unnecessary requests/updates for all value updates and not just from `submit`